core/include/kernel/tee_ta_manager.h

```diff
@@ -93,6 +94,24 @@ struct tee_ta_session {
 	bool unlink;		/* True if session is to be unlinked */
 };
 
+#if defined(CFG_TA_STATS)
+#define MAX_DUMP_SESS_NUM	(16)
+struct tee_ta_dump_stats {
+	TEE_UUID uuid;
```

Comment:
You're providing a bit too much information here, `uuid`, `panicked`, and `heap` should be enough.

Reply:
will remove other dumping info.

Reply:
besides `uuid`, `panicked`, and `heap`, I also keep `is_user_ta`, `sess_num` and its `sess_id`. Those can tell client that how many session are running with this user TA.

Reply:
I don't see how `is_user_ta` or `sess_id` can be of any value to a user space application.

Reply:
I think user can tell if it's user ta first and then print its heap stats. 
Adding sess_id into `tee_ta_dump_stats` is mainly because I need it to get TA's heap stats after `tee_ctxes`/`open_sessions` iteration when calling `tee_ta_get_session(sess_id)`.

Reply:
I remove `sess_id` but keep `is_user_ta` as it can tell client that it's user ta. Only when it's user ta, there is heap stats available to print.
Please see below log info that right now we provide from OPTEE-OS.

tee_mem_stats -ta
TA runtime context dump instance(3):
#1:     TA_UUID(7011a688-ddde-4053-a5a97b3c4ddf13b8)
        panicked:0 -- True if TA has panicked
        session number:0
        ----------------------------------------------------------
#2:     TA_UUID(d96a5b40-e2c7-b1af-87941002a5d5c61b)
        panicked:0 -- True if TA has panicked
        session number:1
        ----------------------------------------------------------
#3:     TA_UUID(8aaaf200-2450-11e4-abe20002a5d5c51e)
        panicked:0 -- True if TA has panicked
        session number:1
        Heap Status:
            allocated:2120(0x848)  max_allocated:2120(0x848) size:65536(0x10000)
            num_alloc_fail:0  biggest_alloc_fail:0
            biggest_alloc_fail_used:0
        ----------------------------------------------------------

Reply:
Why should you return information for anything but user TAs? It seems that `is_user_ta` will always be true.